### ISSUE TITLE
noUncheckedIndexedAccess and checked bracket notation access

### ISSUE THREAD

**Issue by PSeitz** [microsoft/TypeScript#43062]
# Bug Report

### 🔎 Search Terms
noUncheckedIndexedAccess bracket access undefined

### 🕗 Version & Regression Information

- I was unable to test this on prior versions because `noUncheckedIndexedAccess` is available since 4.1

### ⏯ Playground Link

```ts repro

// @noUncheckedIndexedAccess
const meObj: { [propName: string]: string } = {}
const meKey: keyof typeof meObj = 'meh'
if (meObj[meKey]){
  meObj[meKey].toString() // Object is possibly 'undefined'.
}
```

[Workbench Repro](https://www.typescriptlang.org/dev/bug-workbench/?#code/LAKA9GAEACB2D2BVWBjAFgUxQawwEwElY8MAPfAQRRQwGdbQV5ZaAXSAWwwHkAjAKwBckAN6QA2gAcATvEkA5AIZdhbaQEtYAcwC6q1hu2QAvpAC8o442ZtOGANIYAnsNxP4AM0isnkjJ7s+fnNIAHIuNFDQdS8ACi4g8S5HJx0AShFQSECBJIdnHQA6VngAZQNNLVi00GMgA)

### 🙁 Actual behavior

Compile Error: Object is possibly 'undefined'. error

### 🙂 Expected behavior

Typescript should be able to recognize the existence of the property with a check. This generates a lot of wrong errors in our codebase when the `noUncheckedIndexedAccess` compiler flag is enabled.

**Comment by MartinJohns** [comment]
Duplicate of #10530. Type narrowing does not occur for indexed access forms `e[k]` where `k` is not a literal.

**Comment by PSeitz** [comment]
This one also has similar examples further below https://github.com/microsoft/TypeScript/issues/41612
I'll close this one then.
